pkg/test/harness.go

```diff
@@ -356,6 +370,34 @@ func (h *Harness) RunTests() {
 	h.T.Log("run tests finished")
 }
 
+// testPreProcessing provides preprocessing bring all tests suites local if there are any refers to URLs
+func (h *Harness) testPreProcessing() []string {
+	testDirs := []string{}
+	// preprocessing step
+	for _, dir := range h.TestSuite.TestDirs {
+		if http.IsURL(dir) {
+			err := h.initTempPath()
+			if err != nil {
+				h.T.Fatal(err)
+			}
+			client := http.NewClient()
+			h.T.Logf("downloading %s", dir)
+			filePath, err := client.DownloadFile(dir, h.tempPath)
```

Comment:
If I read the code correctly we may have a problem when we encounter sth like this:

```
TestDirs:
  - http://someHost/suiteA/tests.tgz
  - http://someHost/suiteB/tests.tgz
```

Reply:
oh yeah... I did have in mind to solve that problem.. thx!